### PR TITLE
Use usethis rather than devtools to create packages and use GitHub

### PR DESCRIPTION
**Pull request by jimhester**
These previously deprecated functions were removed in devtools 2.1.0

### CODE REVIEW

**Comment by jimhester** [comment]
devtools 2.1.0 is not yet on CRAN, but I plan to submit it today.

File: R/discovery.R [hunk 1]
```diff
@@ -77,7 +78,9 @@ gar_create_package <- function(api_json,
   
   if(github){
     ## if fail, use git add
-    devtools::use_github(pkg = package_dir, protocol = "https")
+    usethis::with_project(package_dir,
```
Comment (MarkEdmondson1234) [inline R/discovery.R]: @jimhester may I ask `with_project()`'s behaviour, will it use the same folder as the issuing RStudio project?

Comment (jimhester) [reply R/discovery.R]: `with_project()` will temporarily set the usethis project to the first argument, then run the code in the second.

Comment (MarkEdmondson1234) [reply R/discovery.R]: Thanks Jim

**Pull request merged**